Fix MongoDB Manager stubs

GromNaN opened the pull request:
Sync some outdated stubs while [implementing rector on `mongodb/mongodb`](https://github.com/mongodb/mongo-php-library/pull/1088).

- `Server::executeBulkWrite` https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Server.stub.php#L74-L79

- `Manager::executeBulkWrite` https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Manager.stub.php#L18-L23

Some outdated definition might still exist, [I made a note to check it all later](https://jira.mongodb.org/browse/PHPORM-23).

GromNaN committed:
Fix MongoDB executeBulkWrite stub

https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Server.stub.php#L74-L79

https://github.com/mongodb/mongo-php-driver/blob/b93f7c2bd10f6f1710aadb0505045df2d221b3ea/src/MongoDB/Manager.stub.php#L18-L23

alcaeus commented on resources/functionMap.php:
```diff
@@ -6774,11 +6774,12 @@
+'MongoDB\Driver\Manager::addSubscriber' => ['void', 'subscriber'=>'MongoDB\Driver\Monitoring\Subscriber'],
+'MongoDB\Driver\Manager::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulk'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
+'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
 'MongoDB\Driver\Manager::executeDelete' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'filter'=>'array|object', 'deleteOptions='=>'array', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
 'MongoDB\Driver\Manager::executeInsert' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'document'=>'array|object', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
```
These two methods don't exist: https://www.php.net/manual/en/class.mongodb-driver-manager.php#mongodb-driver-manager.synopsis

GromNaN replied:
Did they ever exist? I think we need a script update this file for the ext-mongodb reference.

alcaeus replied:
No, these methods never existed.

alcaeus commented on resources/functionMap.php:
```diff
@@ -6774,11 +6774,12 @@
-'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Manager::addSubscriber' => ['void', 'subscriber'=>'MongoDB\Driver\Monitoring\Subscriber'],
+'MongoDB\Driver\Manager::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulk'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
+'MongoDB\Driver\Manager::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
```
Note, there are also `executeReadCommand`, `executeReadWriteCommand`, and `executeWriteCommand`.

GromNaN replied:
Updated

alcaeus commented on resources/functionMap.php:
```diff
@@ -6774,11 +6774,12 @@
 'MongoDB\Driver\Manager::executeInsert' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'document'=>'array|object', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
-'MongoDB\Driver\Manager::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Manager::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
 'MongoDB\Driver\Manager::executeUpdate' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'filter'=>'array|object', 'newObj'=>'array|object', 'updateOptions='=>'array', 'writeConcern='=>'MongoDB\Driver\WriteConcern'],
```
This method does not exist.

GromNaN replied:
Removed

alcaeus commented on resources/functionMap.php:
```diff
@@ -6794,9 +6795,9 @@
-'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'zwrite'=>'BulkWrite'],
-'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
-'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'bulkWrite'=>'BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|nullarray|MongoDB\Driver\WriteConcern|null'],
```
I think the `bulkWrite` argument needs a FQCN. There's also a missing return type

```suggestion
'MongoDB\Driver\Server::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
```

GromNaN replied:
Fixed

alcaeus commented on resources/functionMap.php:
```diff
@@ -6794,9 +6795,9 @@
-'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'bulkWrite'=>'BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|nullarray|MongoDB\Driver\WriteConcern|null'],
+'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
+'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|RMongoDB\Driver\ReadPreference|null'],
```
An extra letter snuck in here:
```suggestion
'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'options='=>'array|MongoDB\Driver\ReadPreference|null'],
```

GromNaN replied:
Fixed

GromNaN committed:
Update MongoDB stubs

Taking ext-mongodb source as reference.
https://github.com/mongodb/mongo-php-driver/blob/master/src/MongoDB/Manager.stub.php

GromNaN reviewed (commented):
Updated `MongoDB\Driver\Manager` using ext-mongodb source stubs: https://github.com/mongodb/mongo-php-driver/blob/master/src/MongoDB/Manager.stub.php

alcaeus commented on resources/functionMap.php:
```diff
@@ -6794,9 +6800,9 @@
-'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'zwrite'=>'BulkWrite'],
-'MongoDB\Driver\Server::executeCommand' => ['MongoDB\Driver\Cursor', 'db'=>'string', 'command'=>'MongoDB\Driver\Command', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
-'MongoDB\Driver\Server::executeQuery' => ['MongoDB\Driver\Cursor', 'namespace'=>'string', 'query'=>'MongoDB\Driver\Query', 'readPreference='=>'MongoDB\Driver\ReadPreference'],
+'MongoDB\Driver\Server::executeBulkWrite' => ['', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|nullarray|MongoDB\Driver\WriteConcern|null'],
```
Return type is still missing, and the type for `options` is invalid:

```suggestion
'MongoDB\Driver\Server::executeBulkWrite' => ['MongoDB\Driver\WriteResult', 'namespace'=>'string', 'bulkWrite'=>'MongoDB\Driver\BulkWrite', 'options='=>'array|MongoDB\Driver\WriteConcern|null'],
```

GromNaN committed:
Update resources/functionMap.php

Co-authored-by: Andreas Braun <alcaeus@users.noreply.github.com>

GromNaN committed:
Merge branch '1.10.x' into mongodb-executebulkwrite

GromNaN commented:
Replaced by https://github.com/phpstan/phpstan-src/pull/2473
